### PR TITLE
Product Switcher documentation and configurable label and icon

### PR DESCRIPTION
**Pull request by maxmarkus**
- label and icon are configurable
- adds documentation

### CODE REVIEW

File: docs/navigation-parameters-reference.md [hunk 1]
```diff
@@ -231,3 +250,16 @@ The profile section is a configurable drop-down list available in the top naviga
     - **url** is the external URL that the link leads to.
 
 
+## Product switcher
+
+The product switcher is a popover with tiles available in the top navigation bar. It allows you to switch between a curated list of navigation elements. To do so, add the **productSwitcher** property to the **navigation** object using the following optional properties:
+
+- **label** defines the label of the product switcher. It is shown as title attribute on hover in the top navigation and as headline in the mobile popover.
```
Comment (bszwarc) [inline docs/navigation-parameters-reference.md]:
```suggestion
- **label** defines the label of the product switcher. It is displayed as a title attribute on hover in the top navigation and as a headline in the mobile pop-up.
```

File: core/examples/luigi-sample-angular/src/luigi-config/extended/navigation.js [hunk 2]
```diff
@@ -272,8 +272,11 @@ class Navigation {
     return items;
   };
 
-  // The following configuration will be used to render a product switcher component
+  // The following configuration will be used to render a product switcher at the end of the top navigation
   productSwitcher = {
+    label: 'My Products',
```
Comment (JohannesDoberer) [inline core/examples/luigi-sample-angular/src/luigi-config/extended/navigation.js]: If there is a empty object `prodcutSwitcher={}` we get the following view on mobile.
Bildschirmfoto 2019-07-31 um 16 58 57

Comment (maxmarkus) [reply core/examples/luigi-sample-angular/src/luigi-config/extended/navigation.js]: Thanks, I have added the default label _My Products_. Is also required for backwards compatibility.

Comment (JohannesDoberer) [reply core/examples/luigi-sample-angular/src/luigi-config/extended/navigation.js]: 👍

**Pull request merged**